icu4c/source/common/charstr.cpp

```diff
@@ -141,6 +141,38 @@ CharString &CharString::append(const char *s, int32_t sLength, UErrorCode &error
     return *this;
 }
 
+CharString &CharString::appendNumber(int32_t number, UErrorCode &status) {
+    if (number < 0) {
+        this->append('-', status);
+        if (U_FAILURE(status)) {
+            return *this;
+        }
+    }
+
+    if (number == 0) {
+        this->append('0', status);
+        return *this;
+    }
+
+    int32_t numLen = 0;
+    while (number != 0) {
```

Comment:
If you change this to a do ... while() loop, so that it always runs at least once, you can get rid of the separate handling of zero, above.

Should number be an int64_t, just in case we want it some day?

Is itoa() safe to use from our code? It seems a shame to have to reinvent the wheel. It's not officially standard, so I don't know.